src/Symfony/Component/Console/Terminal.php

```diff
@@ -43,15 +51,23 @@ public static function getTermColorSupport(): AnsiColorMode
             $term = strtolower($term);
 
             if (str_contains($term, 'truecolor')) {
-                return AnsiColorMode::Ansi24;
+                return self::setTermColorMode(AnsiColorMode::Ansi24);
             }
 
             if (str_contains($term, '256color')) {
-                return AnsiColorMode::Ansi8;
+                return self::setTermColorMode(AnsiColorMode::Ansi8);
             }
         }
 
-        return AnsiColorMode::Ansi4;
+        return self::setTermColorMode(self::DEFAULT_COLOR_MODE);
+    }
+
+    /**
+     * Force a terminal color mode rendering.
+     */
+    public static function setTermColorMode(?AnsiColorMode $colorMode): ?AnsiColorMode
```

Comment:
Returning the color mode passed as argument makes no sense to me in a public API.

Reply:
Ok, I will change that. It was just a less verbose way to deal with the code in `getTermColorMode()`.

Another possibility is to remove completely the setTermColorMode and make the `$termColorMode ` property public?

Reply:
Making a mutable property public would mean that we will **never** be able to refactor this code (as we won't be able to provide a BC layer).

The code in `getTermColorMode` can assign the property directly. But changing the public API of the class to simplify the internal implementation of the other method is bad (the public API is the most important part of the class)

Reply:
Done :)